tests/storage/linstor/test_linstor_sr.py

```diff
@@ -136,17 +136,27 @@ def test_linstor_missing(self, linstor_sr, host):
 # --- Test diskless resources --------------------------------------------------
 
 def _get_diskful_hosts(host, controller_option, volume_name):
-    # Find host where volume is diskless
-    # | {volume_name} | {host} | 7017 | Unused | Ok    |   UpToDate | 2023-10-24 18:52:05 |
-    lines = host.ssh([
-        "linstor", controller_option, "resource", "list",
-        "|", "grep", volume_name, "|", "grep", "UpToDate"
-    ]).splitlines()
-    diskfuls = []
-    for line in lines:
-        hostname = line.split('|')[2].strip()
-        diskfuls += hostname
-    return diskfuls
+    attempt = 0
+    retries = 3
+    sleep_sec = 5
+    while attempt < retries:
+        try:
+            # Find host where volume is diskless
+            # | {volume_name} | {host} | 7017 | Unused | Ok    |   UpToDate | 2023-10-24 18:52:05 |
+            lines = host.ssh([
+                "linstor", controller_option, "resource", "list",
+                "|", "grep", volume_name, "|", "grep", "UpToDate"
+            ]).splitlines()
+            diskfuls = []
+            for line in lines:
+                hostname = line.split('|')[2].strip()
+                diskfuls += hostname
+            return diskfuls
+        except SSHCommandFailed:
+            attempt += 1
```

Comment:
I think it's not entirely ok in this case: is it acceptable to return only one diskful when there is, for example, another replica with a temporary creation status? Should we wait for a status change to potentially return two diskfuls?

Reply:
That can be a TODO.

In a situation where some node(s) might have rebooted, there can be status of `unknown` too, so perhaps more awaiting to change. Since this function was specific to return `UpToDate`, I think it's ok to retry SSH and return currently available `UpToDate`.

We can look at writing more mature function that can preempt certain status and wait intelligently. Do you prefer to do it now or after this is through?

Reply:
@rushikeshjadhav UP, can you add the todo? I would like to have this merged.